xpublish_edr/plugin.py

```diff
@@ -79,65 +93,71 @@ def get_position(
             Extra selecting/slicing parameters can be provided as extra query parameters
             """
             try:
-                ds = dataset.cf.sel(X=query.point.x, Y=query.point.y, method="nearest")
+                ds = select_postition(dataset, query.geometry)
             except KeyError:
                 raise HTTPException(
                     status_code=404,
                     detail="Dataset does not have CF Convention compliant metadata",
                 )
 
-            if query.z:
-                ds = dataset.cf.sel(Z=query.z, method="nearest")
+            logger.debug(
+                f"Dataset filtered by position ({query.geometry.x}, {query.geometry.y}): {ds}",
+            )
 
-            if query.datetime:
-                datetimes = query.datetime.split("/")
+            try:
+                ds = select_query(ds, query, dict(request.query_params))
```

Comment:
```suggestion
                ds = select_by_query_params(ds, query, dict(request.query_params))
```

Reply:
I'd also be tempted to reshuffle `select_query` to be a method on `EDRQuery`.
```suggestion
                ds = edr_query.select(ds, dict(request.query_params))
```